tools/caffe.cpp

```diff
@@ -303,7 +319,17 @@ int time() {
 
   // Set device id and mode
   vector<int> gpus;
+#ifndef CPU_ONLY
+  // Read flags for list of GPUs
   get_gpus(&gpus);
+  while (gpus.size() > 1) {
```

Comment:
This reads a bit oddly to me... can't you just reconstruct the `vector` as a singleton (using its iterator)? And why bother to print the GPUs that we _aren't_ using?